### PR TITLE
Making sure `web.Ctx.CookieDomain()` does not include port.

### LINKED ISSUE #156: v3: `web.Ctx.CookieDomain()` includes port
This leads to

```
...
2019-12-16 11:17:14.441421 I | net/http: invalid Cookie.Domain "localhost:8080"; dropping domain attribute
...
```

See [an example][1], we are [using][2] `u.Host` rather than `u.Hostname()`:

```
=========================================
u.Host:       "localhost:8080"
u.Hostname(): "localhost"
Raw URL:      "http://localhost:8080"
Parsed URL:   &url.URL{Scheme:"http", Opaque:"", User:(*url.Userinfo)(nil), Host:"localhost:8080", Path:"", RawPath:"", ForceQuery:false, RawQuery:"", Fragment:""}
=========================================
u.Host:       "web.invalid"
u.Hostname(): "web.invalid"
Raw URL:      "https://web.invalid"
Parsed URL:   &url.URL{Scheme:"https", Opaque:"", User:(*url.Userinfo)(nil), Host:"web.invalid", Path:"", RawPath:"", ForceQuery:false, RawQuery:"", Fragment:""}
```

(It's going to take me less time to write a fix than it did filing this.)

[1]: https://play.golang.org/p/kIgWUP_k1qu
[2]: https://github.com/blend/go-sdk/blob/480ad70d161406204ef58d3528b0ac17cf64bf0a/web/ctx.go#L262-L267

### PR DESCRIPTION
**Pull request by dhermes**
## PR Summary

Fixes #156

 - **Type:** Bugfix
 - **Issue Link:** https://github.com/blend/go-sdk/issues/156
 - **Intended Change Level:** patch

### CODE REVIEW

File: web/ctx.go [hunk 1]
```diff
@@ -261,9 +262,12 @@ func (rc *Ctx) PostBodyAsXML(response interface{}) error {
 // CookieDomain returns the cookie domain for a request.
 func (rc *Ctx) CookieDomain() string {
 	if rc.App != nil && rc.App.Config.BaseURL != "" {
-		return webutil.MustParseURL(rc.App.Config.BaseURL).Host
+		u := webutil.MustParseURL(rc.App.Config.BaseURL)
```
Comment (wcharczuk) [inline web/ctx.go]: why not just `s/.Host/.Hostname()`?

File: web/ctx.go [hunk 2]
```diff
@@ -261,9 +262,12 @@ func (rc *Ctx) PostBodyAsXML(response interface{}) error {
...
+		return u.Hostname()
 	}
-	return rc.Request.Host
+	// Strip off the port if the hostname includes it.
+	parts := strings.Split(rc.Request.Host, ":")
```
Comment (dhermes) [inline web/ctx.go]: Thoughts on just doing what `net/url` does [in `splitHostPort`][1]?

[1]: https://github.com/golang/go/blob/go1.13.5/src/net/url/url.go#L1071

File: web/ctx.go (hunk 1 above)
Comment (dhermes) [reply web/ctx.go]: See my comment on https://github.com/blend/go-sdk/issues/156#issue-538615678. TL;DR is [they are different][1]

```
=========================================
u.Host:       "localhost:8080"
u.Hostname(): "localhost"
Raw URL:      "http://localhost:8080"
Parsed URL:   &url.URL{Scheme:"http", Opaque:"", User:(*url.Userinfo)(nil), Host:"localhost:8080", Path:"", RawPath:"", ForceQuery:false, RawQuery:"", Fragment:""}
=========================================
u.Host:       "web.invalid"
u.Hostname(): "web.invalid"
Raw URL:      "https://web.invalid"
Parsed URL:   &url.URL{Scheme:"https", Opaque:"", User:(*url.Userinfo)(nil), Host:"web.invalid", Path:"", RawPath:"", ForceQuery:false, RawQuery:"", Fragment:""}
```

[1]: https://play.golang.org/p/kIgWUP_k1qu

File: web/ctx.go (hunk 2 above)
Comment (dhermes) [reply web/ctx.go]: We could make it even smaller though because we wouldn't need to check if `:port` is valid?

Comment (dhermes) [reply web/ctx.go]: Ooooh, TIL `net.SplitHostPort` exists ([example][1]) but it **requires** a port:

```
=================================
hostport: "localhost:8080"
host:     "localhost"
port:     "8080"
err:      <nil>
=================================
hostport: "web.invalid"
host:     ""
port:     ""
err:      &net.AddrError{Err:"missing port in address", Addr:"web.invalid"}
=================================
hostport: "[::1]:16789"
host:     "::1"
port:     "16789"
err:      <nil>
```

[1]: https://play.golang.org/p/kA0eZlp8uJT

**Pull request merged**